### PR TITLE
Var to let in core

### PR DESCRIPTION
**Pull request by hoodmane**
PR #979 replaces `var` ==> `let`. Turns out this is really hard, so I reverted the EM_JS in that PR to using `var`. It still doesn't pass the tests on a webworker though. This PR does var ==> let on the EM_JS. Hopefully it will be a bit easier?

The surprisingly subtle and frustrating bugs that are making this difficult is further proof that `var` needs to be replaced.

This was done with:
`find src -type f -name "*.c"  -or -name "*.h" | xargs sed -E -i 's/\bvar\b/let/g'`

### CODE REVIEW

File: src/core/error_handling.h [hunk 1]
```diff
@@ -32,27 +32,46 @@ log_error(char* msg);
  */
 
 // clang-format off
+#ifdef DEBUG_F
```
Comment (rth) [inline src/core/error_handling.h:35]: There are unrelated code changes here which should be removed. Otherwise yes, splitting that PR in smaller chunks in the right approach and this LGTM.

Comment (hoodmane) [reply src/core/error_handling.h:35]: See: "Check out error_handling.h and Makefile from var-to-let" this was intended to improve the error messages generated.
